### PR TITLE
xds-k8s README.md: improve basic setup docs

### PR DESCRIPTION
**Pull request by sergiitk**
Reorder sections, recommend local-dev.cfg instead of grpc-testing to avoid confusion.

### CODE REVIEW

**Review by easwars — APPROVED** [review]

**Comment by sergiitk** [comment]
also cc @murgatroid99

File: tools/run_tests/xds_k8s_test_driver/README.md [hunk 1]
```diff
@@ -269,7 +227,44 @@ envsubst < config/local-dev.cfg.example > config/local-dev.cfg
...
+
+Test suite meant to confirm that basic xDS features work as expected. Executing
+it before other test suites will help to identify whether test failure related
+to specific features under test, or caused by unrelated infrastructure
+disturbances.
+
+```shell
+# Help
+python -m tests.baseline_test --help
+python -m tests.baseline_test --helpfull
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/README.md:244]: s/python/python3/g

File: tools/run_tests/xds_k8s_test_driver/README.md [hunk 2]
```diff
@@ -269,7 +227,44 @@ envsubst < config/local-dev.cfg.example > config/local-dev.cfg
...
+
+# Run the baseline test with local-dev.cfg settings
+python -m tests.baseline_test --flagfile="config/local-dev.cfg"
+  
+# Same as above, but using the helper script
+./run.sh tests/baseline_test.py
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/README.md:250]: optional: I would recommend to layout all the command line arguments, so the people onboarding the framework can see what's getting configured here. The `./run.sh` can exist as a shortcut documented as an alternative way of running the tests.

File: tools/run_tests/xds_k8s_test_driver/README.md (hunk 1 above)
Comment (sergiitk) [reply tools/run_tests/xds_k8s_test_driver/README.md:244]: This assumes virtualenv is active, so `python` will point to the correct version.

File: tools/run_tests/xds_k8s_test_driver/README.md (hunk 2 above)
Comment (sergiitk) [reply tools/run_tests/xds_k8s_test_driver/README.md:250]: People onboarding are likely to use `./run.sh` - that's from my experience of a few folks recently. Regarding command arguments, there are a lot of them. That's why I'm suggesting `--help` above.

But I think we could list all *required* arguments. Would it be ok if this is tackled outside of this PR?

**Review by lidizheng — APPROVED** [review]

**Pull request merged**